### PR TITLE
Intial draft for maintainer doc

### PR DESCRIPTION
**Pull request by Sherwin-14**
I've made this doc as per my understanding. Open to discussing what more/less could be done here.

----
📚 Documentation preview 📚: https://earthaccess--608.org.readthedocs.build/en/608/

### CODE REVIEW

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:58]: Let's re-use our existing releasing page `contributing/releasing.md` instead :)

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:5]: Let's reserve header level 1 for the title of the document, and add one extra header level to all the other sub-headers.

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:11]:
```suggestion
Also, if you're interested in helping managing issues with labels and interacting with incoming requests, you can have a "triager" role!

To get permissions, please start by participating on GitHub by answering questions, reviewing PRs, or contributing code or documentation. Once you're feeling comfortable, you can ask any of our maintainers for permissions by `@`ing them on GitHub.
```

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:9]:
```suggestion
If you are interested in becoming a maintainer, you can join our community. Maintainers have several important responsibilities, so please read on to understand the role.
```

I feel the "actively involved" part is more clearly explained in the next section :)

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:36]: The above two sections are exactly what we need! Thank you :star_struck:

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:42]:
```suggestion
```

Our `gh-pages` branch is static and only redirects to ReadTheDocs, so we can remove this section.

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:46]:
```suggestion
The GitHub Actions CI services handle the project's building, testing, and management across Linux, macOS, and Windows platforms. The CI workflow manages the continuous integration processes, and list out their respective dependencies. Here we briefly summarize the functions of the files in CI.
```

TODO: Link to the CI workflow YAML file being referenced here

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:44]: We could call this "Continuous Integration & Delivery". We have quite a few workflows going on, and I don't think we need to go in depth on all of them right now, we can always build that out more later. But I do think it's important to call out that delivery / releasing is part of our workflows :)

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:50]:
```suggestion
```environment-mindeps.yaml``` - Specifies an environment with the minimum supported dependency versions to detect backwards-incompatible code.
```

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:54]:
```suggestion
[ReadTheDocs](https://readthedocs.org/projects/earthaccess/) is used to generate and host [our documentation website](https://earthaccess.readthedocs.io/) as well as preview documentation changes made in pull requests. This service uses a configuration file, ```.readthedocs.yaml```.
```

File: docs/maintainers-guide.md
Comment (mfisher87) [inline docs/maintainers-guide.md:48]:
```suggestion
`environment-dev.yml`  - The environment configuration file specifies dependencies for Python development.
```

Please change any inline code to single backticks throughout :bow:

**Review by mfisher87 — CHANGES_REQUESTED** [review]
This looks great! Thanks so much for your work. A couple nitpicky suggestions :)

File: docs/maintainers-guide.md
Comment (Sherwin-14) [reply docs/maintainers-guide.md:36]: Could you please be more specific about this? I'm kind of confused here.

Comment (mfisher87) [reply docs/maintainers-guide.md:36]: Sorry for being unclear. I was saying that I think the "Maintainer Responsibilities and Expectations" & " Maintainer Processes Beyond Regular Contributing" sections look great :)

**Comment by mfisher87** [comment]
pre-commit.ci autofix

**Review by mfisher87 — COMMENTED** [review]
Looks great, just one last thing: There's currently no link to get to this page. What do you think of including it in the "Work with us" part of the side nav (`mkdocs.yml`)?

I was thinking we should also move the file into the `doc/contributing` directory. Do you agree?

**Review by mfisher87 — APPROVED** [review]

**Comment by mfisher87** [comment]
@Sherwin-14 LGTM! Let me know if you're ready to merge or if you'd like some time to rebase beforehand.

**Comment by Sherwin-14** [comment]
@mfisher87 I am ready to merge, you can decide as per your convenience :)

**Comment by mfisher87** [comment]
Thank you for your contribution :)

**Pull request merged**
